### ISSUE TITLE
FirebaseError: Firebase: No Firebase App '[DEFAULT]' has been created - call Firebase App.initializeApp() (app/no-app).

### ISSUE THREAD

**Issue by kylesmith13** [firebase/firebase-js-sdk#6086]
### [REQUIRED] Describe your environment

  * Operating System version: Montery 12.0
  * Browser version: Chrome 99.0.4844.74, tried safari as well
  * Firebase SDK version: 9.6.9, but tried other versions
  * Firebase Product: the entire product, but trying to use messaging

#### Steps to reproduce:
Last night I got this code to successfully work and it was spitting out my client ID and asking if i'd like to allow notifications. My computer crashed and not matter what I do, it no longer works. I tried changing versions, i made a new nextjs app and moved some files over, same error. I loaded up React-Firebase-Cloud-Messaging-Demo and put my credentials in (i'm using this as an example to build mine) and it works fine. It worked fine last night. I'm so baffled at how this doesn't work anymore and no matter what I try to do it spits out the same error.

#### Relevant Code:

```javascript
// Firebase Cloud Messaging Configuration File.
// Read more at https://firebase.google.com/docs/cloud-messaging/js/client && https://firebase.google.com/docs/cloud-messaging/js/receive

import { initializeApp } from 'firebase/app';
import { getMessaging, getToken, onMessage } from 'firebase/messaging';

const firebaseConfig = {
  [REDACTED]
};

initializeApp(firebaseConfig)

const messaging = getMessaging();

export const requestForToken = () => {
  return getToken(messaging, { vapidKey: REDACTED })
    .then((currentToken) => {
      if (currentToken) {
        console.log('current token for client: ', currentToken);
        // Perform any other neccessary action with the token
      } else {
        // Show permission request UI
        console.log('No registration token available. Request permission to generate one.');
      }
    })
    .catch((err) => {
      console.log('An error occurred while retrieving token. ', err);
    });
};
```
basically just importing this into a file and using requestForToken (it never hits this function because it fails at messaging = getMessaging().

`FirebaseError: Firebase: No Firebase App '[DEFAULT]' has been created - call Firebase App.initializeApp() (app/no-app).`

i even tried to set const app = initializeApp(firebaseConfig) and pass it into const messaging = getMessaging(app) and it complains about not being on a support browser, but the other project runs just fine so obviously that's not the case. 
<img width="583" alt="Screen Shot 2022-03-19 at 12 22 16 PM" src="https://user-images.githubusercontent.com/30356483/159131561-6bfa6ffb-cfa1-463b-8743-1786d727277e.png">

**Comment by jbalidiong** [comment]
Thanks for reaching out, @kylesmith13. I tried replicating, but I wasn't able to reproduce the error. I just tried the basic reactjs app (I followed https://reactjs.org/docs/create-a-new-react-app.html). For your reference:

App.js

```
import {initializeApp} from 'firebase/app';
import {getMessaging, getToken, onMessage} from 'firebase/messaging';

const firebaseConfig = {
  apiKey: "",
  authDomain: "",
  databaseURL: "",
  projectId: "",
  storageBucket: "",
  messagingSenderId: "",
  appId: "",
  measurementId: ""
};

initializeApp(firebaseConfig);
const messaging = getMessaging();

// Get registration token. Initially this makes a network call, once retrieved
// subsequent calls to getToken will return from cache.
getToken(messaging, { vapidKey: 'VAPID_KEY' }).then((currentToken) => {
  if (currentToken) {
    // Send the token to your server and update the UI if necessary

    console.log(currentToken);
  } else {
    // Show permission request UI
    console.log('No registration token available. Request permission to generate one.');
    // ...
  }
}).catch((err) => {
  console.log('An error occurred while retrieving token. ', err);
  // ...
});

onMessage(messaging, (payload) => {
  console.log('Message received. ', payload);
  // ...
});
```

public/firebase-messaging-sw.js

```
// Scripts for firebase and firebase messaging
importScripts('https://www.gstatic.com/firebasejs/9.6.8/firebase-app-compat.js');
importScripts('https://www.gstatic.com/firebasejs/9.6.8/firebase-messaging-compat.js');

// Initialize the Firebase app in the service worker by passing the generated config
var firebaseConfig = {
    apiKey: "",
    authDomain: "",
    projectId: "",
    storageBucket: "",
    messagingSenderId: "",
    appId: "",
    measurementId: ""
};

firebase.initializeApp(firebaseConfig);

// Retrieve firebase messaging
const messaging = firebase.messaging();

messaging.onBackgroundMessage(function (payload) {
    console.log('Received background message ', payload);
    const notificationTitle = payload.notification.title;
    const notificationOptions = {
        body: payload.notification.body,
    };

    self.registration.showNotification(notificationTitle,
        notificationOptions);
});

```

Seems like this will turn out to be a timing issue, and that `getMessaging()` is likely getting invoked before your `initializeApp()` code. Would you mind trying to create a blank app, follow the implementation and see if the issue persists?

**Comment by kylesmith13** [comment]
`I tried changing versions, i made a new nextjs app and moved some files over, same error.` i did this on the day of making the issue

**Comment by jbalidiong** [comment]
I'll be closing this for now. If the issue persists, feel free to comment and reopen this issue.
